File: day01/puzzle01.js
let prevLine = "";
let prevValue;
let increases = 0;

process.stdin.on("data", (chunk) => {
  const lines = chunk.toString().split("\n");
  lines[0] = prevLine + lines[0];
  prevLine = lines.pop();
  lines.forEach(processLine);
});

process.stdin.on("close", () => {
  processLine(prevLine);
  console.log("\n\n %d increases", increases);
});

function processLine(line) {
  const value = parseInt(line, 10);
  if (isNaN(value)) {
    return;
  }

  if (prevValue == null) {
    console.log("%d (N/A - no previous measurement)", value);
  } else {
    const didIncrease = value > prevValue;
    if (didIncrease) {
      increases++;
      console.log("%d (increased)", value);
    } else {
      console.log("%d (decreased)", value);
    }
  }

  prevValue = value;
}

File: day01/puzzle01.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	var prevValue *int64
	var increases int64

	for scanner.Scan() {
		value, err := strconv.ParseInt(scanner.Text(), 10, 32)

		if err != nil {
			continue
		}

		if prevValue != nil {

			increased := value > *prevValue
			decreased := value < *prevValue

			if increased {
				increases++
				fmt.Printf("%d: increased\n", value)
			} else if decreased {
				fmt.Printf("%d: decreased\n", value)
			}

		}

		prevValue = &value
	}

	fmt.Printf("\n\n\n%d increases\n", increases)
}
